### PR TITLE
DS-2625 Extendable control panel

### PR DESCRIPTION
**Pull request by kosarko**
https://jira.duraspace.org/browse/DS-2625

### CODE REVIEW

File: dspace-xmlui/src/main/webapp/static/js/controlpanel.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*global jQuery */
```
Comment (mwoodiupui) [inline dspace-xmlui/src/main/webapp/static/js/controlpanel.js]: This file needs a license header.

**Comment by helix84** [comment]
+1, tested in Mirage and Mirage2

File: dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/administrative/controlpanel/ControlPanelHarvestingTab.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    public void addBody(Map objectModel, Division div) throws WingException,
+            SQLException
+    {
+        List harvesterControls = div.addList("oai-harvester-controls",
+                List.TYPE_FORM);
+        harvesterControls.setHead(T_harvest_scheduler_head);
+        harvesterControls.addLabel(T_harvest_label_status);
+        Item status = harvesterControls.addItem();
+        status.addContent(HarvestScheduler.getStatus());
+        status.addXref(contextPath + "/admin/panel?harvest", "(refresh)");
```
Comment (helix84) [inline dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/administrative/controlpanel/ControlPanelHarvestingTab.java]: change to /admin/panel?tab=Harvesting
add 18n, please

**Comment by kosarko** [comment]
Hi, I've rebased DS-2625 to the current master and resolved the conflicts (hopefully the changed propagated here). I haven't tested it as I am having some trouble with vagrant-dspace.

**Comment by KevinVdV** [comment]
I took a quick glance at the code & noticed that the configuration is handled by using dspace plugin mechanism, would it not be better to use spring to configure the control panel ?
I also noticed that a static javascript file is used, does this javascript also work with Mirage 2 (haven't tested it yet though).

**Comment by kosarko** [comment]
Will there be any benefits as a result of using Spring?

I haven't tested it with Mirage 2, but Helix has (see his comment before the rebase) and it was working. The javascript is there just to toggle hide/show on some extra information.

File: dspace/config/modules/controlpanel.cfg [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#---------------------------------------------------------------#
+#------------ADMIN CONTROL PANEL CONFIGURATIONS-----------------#
+#---------------------------------------------------------------#
+
+### Control Panel Tabs
+controlpanel.tabs = Java Information, \
```
Comment (KevinVdV) [inline dspace/config/modules/controlpanel.cfg]: The panel headers cannot be internationalized, best to use a generic identifier & generate the xmlui message using that as opposed to placing them in config.

**Comment by KevinVdV** [comment]
Using spring instead of the plugin classes is strongly recommended since we are going to retire the plugin mechanism in the future.
For this pull request to be accepted you don't have to make the change, it is just a recommend to do it that way. In later releases these plugins will be ported to spring so this feature would be more future proof in that case.

**Comment by KevinVdV** [comment]
* Seeing some changes made to the "CurrentActivityAction" class, although these changes appear to make sense they should be removed from this pull request & placed in a separate one to improve the tracking of event in the control panel.
* The oai.cfg has been adjusted, prob not needed in this pull request.

Aside from these issues the pull request works as intended, so if these 2 can get fixed it has my +1.

**Comment by kosarko** [comment]
> Using spring instead of the plugin classes is strongly recommended since we are going to retire the plugin mechanism in the future.

Wouldn't it be reasonable to mark the PluginManager class deprecated first?

> The oai.cfg has been adjusted, prob not needed in this pull request.

The harvesting tab shows the url in the "Generator settings", `dspace.oai.url` was never filled in so this field was always empty. That's why it was added, but I see some recent changes in fbde10802, so where does one take the oai url?

btw oai.cfg still contains this line:
```
If using oai, the dspace.oai.uri config value must be set.
```

Thanks for the comments, I'll try to have a look into the internationalization and CurrentActivityAction before the year ends...

**Comment by mwoodiupui** [comment]
Any updates? We would like to merge or defer to 7.0 by 20-Jan-2016.

**Comment by kosarko** [comment]
I've added the internationalization for tab links/names, removed the oai.cfg change and squashed the commits. I've also factored the CurrentActivityAction and related changes into a commit of it's own (so you can take it or leave it easily...)

**Comment by KevinVdV** [comment]
Reviewed the latest changes, my comments were properly resolved. 
So a +1 from me, if anybody else can give this a quick spin we can merge it.

**Comment by tdonohue** [comment]
+1, code looks good, and I gave it a quick test and it all works fine.

@kosarko , I think the last thing we'd need (sometime prior to the final 6.0 release) is some final Documentation that we can copy over into the official Docs area (or I can even give you direct access to the official docs to add it in).  I know you started some at: https://wiki.duraspace.org/display/DSPACE/Extensible+control+panel , but it seems to be unfinished (at least the part about adding new tabs, which seems to end abruptly and doesn't explain things like changing configuration or i18n for new tabs).

In any case though, the code itself is good and works. Merging, as this is now at +3 votes.

**Pull request merged**